openedx/core/djangoapps/credit/api/eligibility.py

```diff
@@ -187,7 +187,7 @@ def get_eligibilities_for_user(username, course_key=None):
     """
     eligibilities = CreditEligibility.get_user_eligibilities(username)
     if course_key:
-        course_key = CourseKey.from_string(unicode(course_key))
+        course_key = course_key_from_string_or_404(unicode(course_key))
```

Comment:
is this an appropriate place to use this helper? Is it only being called by a view?